testing/file_test/autoupdate.cpp

```diff
@@ -222,30 +236,30 @@ auto AutoupdateFileTest(
 
   // Stitch together content.
   llvm::SmallVector<const FileTestLineBase*> new_lines;
-  for (auto [filename, non_check_file, stdout_check_file, stderr_check_file] :
-       llvm::zip(filenames, non_check_lines, stdout_check_lines,
-                 stderr_check_lines)) {
+  for (auto [file_number, filename, non_check_file] :
+       llvm::enumerate(filenames, non_check_lines)) {
     llvm::DenseMap<int, int> output_line_remap;
+    llvm::SmallVector<CheckLine*> check_lines_this_file;
     int output_line_number = 0;
-    auto* stdout_check_line = stdout_check_file.begin();
-    auto* stderr_check_line = stderr_check_file.begin();
 
     // Add all check lines from the given vector until we reach a check line
     // attached to a line later than `to_line_number`.
-    auto add_check_lines = [&](const llvm::SmallVector<CheckLine>& lines,
+    // Explicit capture of file_number is a workaround for the inability for
+    // lambdas to implicitly capture a structured binding.
+    auto add_check_lines = [&, file_number = static_cast<int>(file_number)](
+                               const llvm::SmallVector<CheckLine>& lines,
                                CheckLine*& line, int to_line_number,
                                llvm::StringRef indent) {
-      for (; line != lines.end() && line->line_number() <= to_line_number;
+      for (; line != lines.end() && line->file_number() <= file_number &&
+             (line->file_number() < file_number ||
+              line->line_number() <= to_line_number);
```

Comment:
Moving `static_cast` to uses:

```suggestion
auto add_check_lines = [&, file_number = file_number](
                               const llvm::SmallVector<CheckLine>& lines,
                               CheckLine*& line, int to_line_number,
                               llvm::StringRef indent) {
      for (; line != lines.end() &&
             line->file_number() <= static_cast<int>(file_number) &&
             (line->file_number() < static_cast<int>(file_number) ||
              line->line_number() <= to_line_number);
```